lib/chef/provider/execute.rb

```diff
@@ -69,6 +69,14 @@ def sensitive?
         !!new_resource.sensitive
       end
 
+      def stream_to_formatter?
+        Chef::Config[:always_stream_execute] || run_context.events.formatter?
```

Comment:
Did we want to make streaming an option on the resource? Seems like `run_context.events.formatter?` will be true whenever you run `chef-client` with default options on the command line; you could disable it with `sensitive(true)` but that hides outputs in the exception case, too, which wouldn't be what you want if you just want to suppress a really verbose command for only the success case.